cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/runtime/batch/DynamicPartitionerWithAvroTest.java

```diff
@@ -114,6 +114,9 @@ public void apply() throws IOException {
           Map<PartitionKey, PartitionDetail> partitions = new HashMap<>();
           for (PartitionDetail partition : pfs.getPartitions(null)) {
             partitions.put(partition.getPartitionKey(), partition);
+            // check that the mapreduce wrote the output partition metadata to all the output partitions
```

Comment:
Do you have a test case for MultipleOutputs? Two PFS outputting a partition, each with different metadata